frontend/src/js/ui/views/generators/generator/index.jsx

```diff
@@ -58,7 +59,7 @@ export default function () {
 		state.config = pickBy(state.config, (val, key) => key === 'seed' || state.gen.config.some((conf) => conf.key === key));
 	};
 
-	let onexport = (type) => {
+	let onexport = async (type) => {
```

Comment:
The modal not closing is related to onexport suddenly being a ``async`` function. Just remove the async and stick to basic promise operations like in the example below 😄